crates/roc_app/src/wasm32.rs

```diff
@@ -1654,124 +1871,165 @@ impl PartialOrd for WriteResult {
-        debug_assert_eq!(self.discriminant, discriminant_WriteResult::Error);
+        debug_assert_eq!(self.discriminant, discriminant_ReadResult::Error);
         use core::borrow::BorrowMut;
         unsafe { self.payload.Error.borrow_mut() }
     }
 
     pub fn is_Error(&self) -> bool {
-        matches!(self.discriminant, discriminant_WriteResult::Error)
+        matches!(self.discriminant, discriminant_ReadResult::Error)
     }
 
-    pub fn is_Wrote(&self) -> bool {
-        matches!(self.discriminant, discriminant_WriteResult::Wrote)
+    pub fn unwrap_Read(mut self) -> roc_std::RocList<u8> {
+        debug_assert_eq!(self.discriminant, discriminant_ReadResult::Read);
+        unsafe { core::mem::ManuallyDrop::take(&mut self.payload.Read) }
+    }
+
+    pub fn borrow_Read(&self) -> &roc_std::RocList<u8> {
+        debug_assert_eq!(self.discriminant, discriminant_ReadResult::Read);
+        use core::borrow::Borrow;
+        unsafe { self.payload.Read.borrow() }
+    }
+
+    pub fn borrow_mut_Read(&mut self) -> &mut roc_std::RocList<u8> {
+        debug_assert_eq!(self.discriminant, discriminant_ReadResult::Read);
+        use core::borrow::BorrowMut;
+        unsafe { self.payload.Read.borrow_mut() }
+    }
+
+    pub fn is_Read(&self) -> bool {
+        matches!(self.discriminant, discriminant_ReadResult::Read)
     }
 }
 
-impl WriteResult {
+impl ReadResult {
     pub fn Error(payload: StreamErr) -> Self {
         Self {
-            discriminant: discriminant_WriteResult::Error,
-            payload: union_WriteResult {
+            discriminant: discriminant_ReadResult::Error,
+            payload: union_ReadResult {
                 Error: core::mem::ManuallyDrop::new(payload),
             },
         }
     }
 
-    pub fn Wrote() -> Self {
+    pub fn Read(payload: roc_std::RocList<u8>) -> Self {
         Self {
-            discriminant: discriminant_WriteResult::Wrote,
-            payload: union_WriteResult { Wrote: () },
+            discriminant: discriminant_ReadResult::Read,
+            payload: union_ReadResult {
+                Read: core::mem::ManuallyDrop::new(payload),
+            },
         }
     }
 }
 
-impl Drop for WriteResult {
+impl Drop for ReadResult {
     fn drop(&mut self) {
         // Drop the payloads
         match self.discriminant() {
-            discriminant_WriteResult::Error => unsafe {
+            discriminant_ReadResult::Error => unsafe {
                 core::mem::ManuallyDrop::drop(&mut self.payload.Error)
             },
-            discriminant_WriteResult::Wrote => {}
+            discriminant_ReadResult::Read => unsafe {
+                core::mem::ManuallyDrop::drop(&mut self.payload.Read)
+            },
         }
     }
 }
 
+impl roc_std::RocRefcounted for ReadResult {
+    fn inc(&mut self) {
+        unimplemented!();
+    }
+    fn dec(&mut self) {
+        unimplemented!();
+    }
+    fn is_refcounted() -> bool {
+        true
+    }
+}
```

Comment:
Same here

Reply:
This was also generated by `RustGlue.roc`. I think we could remove it if we wanted to, because it's not used or needed. I think the implementation is just to satisfy the trait requirement.

Reply:
Thanks, I'll try to remove them all